### PR TITLE
Show excludes in Excel report

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Visualize excluded projects and scopes in the Excel report. See the commit messages for details.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/883)

### CODE REVIEW

File: reporter/src/main/kotlin/reporters/ExcelReporter.kt [hunk 1]
```diff
@@ -323,9 +448,10 @@ private fun XSSFSheet.finalize(headerRows: Int, totalRows: Int, totalColumns: In
 
 private const val ELLIPSIS = "[...]"
 
-private fun Collection<*>.joinWithLimit(
+private fun <T> Iterable<T>.joinWithLimit(
         separator: String = " \n",
-        maxLength: Int = 32767 /* This is the maximum Excel cell content length. */
-) = joinToString(separator).let { cellContent ->
+        maxLength: Int = 32767, /* This is the maximum Excel cell content length. */
+        transform: ((T) -> CharSequence)? = null
```
Comment (sschuberth) [inline reporter/src/main/kotlin/reporters/ExcelReporter.kt]: Nit: This new parameter seems to be never used. Will it be in future commits?

File: reporter/src/main/kotlin/reporters/ExcelReporter.kt [hunk 2]
```diff
@@ -323,9 +448,10 @@ private fun XSSFSheet.finalize(headerRows: Int, totalRows: Int, totalColumns: In
 
 private const val ELLIPSIS = "[...]"
 
-private fun Collection<*>.joinWithLimit(
+private fun <T> Iterable<T>.joinWithLimit(
```
Comment (sschuberth) [inline reporter/src/main/kotlin/reporters/ExcelReporter.kt]: Should this maybe already be done in the first commit? In other words, is the intermediate change to `Collection<*>` meaningful?

File: model/src/main/kotlin/config/Excludes.kt [hunk 3]
```diff
@@ -77,4 +77,10 @@ data class Excludes(
     fun isScopeExcluded(scope: Scope, project: Project) =
             scopes.any { it.matches(scope.name) }
                     || findProjectExclude(project)?.scopes?.any { it.matches(scope.name) } ?: false
+
+    /**
+     * Map the project excludes by the identifiers of the provided [projects].
+     */
+    fun projectExcludesById(projects: Set<Project>) =
```
[the remark this answers is not in the document]
Comment (mnonnenmacher) [reply model/src/main/kotlin/config/Excludes.kt]: Done.

File: reporter/src/main/kotlin/reporters/ExcelReporter.kt (hunk 1 above)
Comment (mnonnenmacher) [reply reporter/src/main/kotlin/reporters/ExcelReporter.kt]: No, removed.

File: reporter/src/main/kotlin/reporters/ExcelReporter.kt (hunk 2 above)
Comment (mnonnenmacher) [reply reporter/src/main/kotlin/reporters/ExcelReporter.kt]: Yes and no :)

**Comment by sschuberth** [comment]
Again, I didn't look deeply into the algorithm, but the formal aspects now look fine to me.

**Pull request merged**